Python/optimizer.c

```diff
@@ -579,14 +607,8 @@ translate_bytecode_to_trace(
             case JUMP_BACKWARD:
             case JUMP_BACKWARD_NO_INTERRUPT:
             {
-                if (instr + 2 - oparg == initial_instr && code == initial_code) {
-                    RESERVE(1);
-                    ADD_TO_TRACE(_JUMP_TO_TOP, 0, 0, 0);
-                }
-                else {
-                    OPT_STAT_INC(inner_loop);
-                    DPRINTF(2, "JUMP_BACKWARD not to top ends trace\n");
-                }
+                OPT_STAT_INC(inner_loop);
+                DPRINTF(2, "JUMP_BACKWARD not to top ends trace\n");
```

Comment:
Isn't it "any JUMP_BACKWARD ends trace" now? (After the initial one, if any.)

Reply:
Maybe also assert that it isn't the top.